### PR TITLE
Make e2e/framework consistently use the corev1 alias

### PR DESCRIPTION
**Pull request by johnSchnake**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
There is inconsistent aliasing of the k8s.io/api/core/v1
package. As part of a refactor of the framework code we
would like to increasing consistency on things like that.

Used the alias corev1 throughout.

**Which issue(s) this PR fixes**:
Ref #77045

**Special notes for your reviewer**:
N/A

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig testing
/area test
/area test-framework
/priority important-longterm

### CODE REVIEW

**Comment by timothysc** [comment]
/hold 

We want to alias overrides that are framework utils, but not core things.  I don't know why we would want todo this in particular.

**Comment by johnSchnake** [comment]
/retest

This is the timeline of this that shows why we want to do this:
 - testing commons is refactoring e2e/framework so I tried to extract a single set of functions out into their own package which resulted in a _huge_ diff because it touched so many files
 - by touching all those files, goimports (a common developer tool) tried to also update a ton of aliases, mostly trying to alias `k8s.io/api/core/v1` as `v1`
 - because that constituted a significant portion of the PR I separated out 'run go imports' into its own PR
 - simultaneously @draveness created #77045 complaining about the lack of consistency; sometimes it is imported with an alias, sometimes not
 - I found running goimports over the codebase that was too large a change so I scoped it down to test/e2e
 - a separate PR where I refactored only a handful of files led to disagreement with @fejta about adding the alias `v1 "k8s.io/api/core/v1"` which is what goimports does automatically, instead it was argued that the best (and common) practice is to alias it as `corev1` because it is more helpful/clear
 - Because it updates the call sites as well as the import line, I argued that at a minimum that should be its own PR (this PR in fact, though more PRs could continue to make the rest of the codebase consistent as well)

TL;DR;
 - Things are currently inconsistent with how that import is aliased.
 - goimports wants to alias it as `v1` but that is not descriptive to devs in any way since we have lots of packages called v1
 - the compromise is to make it consistent _and_ with a helpful alias, `corev1`

If this doesnt go in, it will not burn the world, but in the handful of refactoring PRs I've put in, I have spent _way_ too much time talking about how/when/if/why we alias this package and whether it is goimports or the editor doing it (it is goimports btw)

**Comment by timothysc** [comment]
So I'm ok with this for the framework, but goimports is not a requirement.  Before we make this change we should try to be consistent.

**Comment by timothysc** [comment]
@dims could you give me a take from code-organization perspective 

/assign @dims

**Comment by johnSchnake** [comment]
Per discussion on https://github.com/kubernetes/kubernetes/issues/77045 and in slack; closing this PR.
